src/main/java/bio/terra/janitor/service/cleanup/flight/ResourceCleanupStep.java

```diff
@@ -42,31 +41,13 @@ public StepResult doStep(FlightContext flightContext) {
       return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL);
     }
     TrackedResourceState state = trackedResource.get().trackedResourceState();
-    switch (state) {
-      case DONE:
-      case ERROR:
-      case READY:
-        // State is DONE ERROR or READY, this should not happen, fail the flight.
-        throw new IllegalStateException(
-            String.format(
-                "Illegal trackedResource state for trackedResource %s, state is %s",
-                trackedResourceId.toString(), state.toString()));
-      case DUPLICATED:
-      case ABANDONED:
-        // State is DUPLICATED or ABANDONED, skip the cleanup.
-        logger.warn(
-            "Skip resource {} cleanup because the state is {} due to a race.",
-            trackedResourceId.toString(),
-            state.toString());
-        return StepResult.getStepResultSuccess();
-      case CLEANING:
-        // TODO(yonghao): Update cleanup log
-        return cleanUp(cloudResourceUid);
-      default:
-        throw new UnsupportedOperationException(
-            String.format(
-                "Unsupported trackedResource state for trackedResource %s, state is %s",
-                trackedResourceId.toString(), state.toString()));
+    if (state.equals(CLEANING)) {
+      return cleanUp(cloudResourceUid);
+    } else {
+      throw new IllegalStateException(
+          String.format(
+              "Illegal trackedResource state for trackedResource %s, state is %s",
+              trackedResourceId.toString(), state.toString()));
```

Comment:
DUPLICATED and ABANDONED are not unsupported states. They are expected. They should result in stepresultsuccess.

Reply:
Done